src/parser/mage/fire/modules/features/Pyroclasm.tsx

```diff
@@ -15,16 +15,22 @@ const debug = false;
 
 class Pyroclasm extends Analyzer {
 
-  buffAppliedEvent = null;
-  beginCastEvent = null;
-  castEvent = null;
   totalProcs = 0;
   usedProcs = 0;
   unusedProcs = 0;
   overwrittenProcs = 0;
-
-  constructor(...args) {
-    super(...args);
+  beginCastEvent!: {
```

Comment:
```suggestion
  beginCastEvent: {
```

! when defining a prop only makes sense for dependencies. You should almost never need it otherwise and avoid using it to make sure you don't get crashes